src/components/UI/Header/SearchBox/SearchBox.jsx

```diff
@@ -1,19 +1,35 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
 import "./SearchBox.css";
 
 export const SearchBox = () => {
-	const onClickSearchHandler = (e) => e.preventDefault();
+	const [query, setQuery] = useState("");
+	const navigate = useNavigate();
+
+	const onChangeSearchHandler = (e) => {
+		setQuery(e.target.value);
+	};
+
+	const passQuery = (e) => {
+		e.preventDefault();
+		navigate(`/productlisting/search?query=${query}`);
+		setQuery("");
+	};
+
 	return (
-		<form className="search_box" method="get">
+		<form onSubmit={passQuery} className="search_box" method="get">
 			<input
```

Comment:
you could have added a sentence on the search page to show what user has searched to improve user experience